### PR TITLE
[clang-tidy] fix incorrect hint for InitListExpr in prefer-member-initializer

### LINKED ISSUE #77684: [clang-tidy] cppcoreguidelines-prefer-member-initializer fix-it produces broken code
The clang-tidy fix-it `cppcoreguidelines-prefer-member-initializer` turns 
```
struct S {
  S() : M{} { M = 0; }
  int M;
};
```
into
```
struct S {
  S() : M0 { }
  int M;
};
```
Tested with main from 01.01.2024 (f502b981b471c04bef1b2a6c581c02b59f931163).

### PR DESCRIPTION
**Pull request by HerrCai0907**
Fixes: #77684.

### CODE REVIEW

File: clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp [hunk 1]
```diff
@@ -203,6 +203,7 @@ void PreferMemberInitializerCheck::check(
     SourceLocation InsertPos;
     SourceRange ReplaceRange;
     bool AddComma = false;
+    bool AddBracket = false;
```
Comment (5chmidti) [inline clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp:206]: Brackets are `[]`, use brace (`{}`).

File: clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/prefer-member-initializer.cpp [hunk 2]
```diff
@@ -616,3 +616,26 @@ class Foo {
...
+};
+struct T { int a; int b; int c; };
+T v;
+struct S3 {
+// CHECK-MESSAGES: :[[@LINE+1]]:21: warning: 'M' should be initialized in a member initializer of the constructor [cppcoreguidelines-prefer-member-initializer]
+  S3() : M{1,2,3} { M = v; }
+// CHECK-FIXES: {{ S3.+M\{v\} }}
+  T M;
+};
+}
```
Comment (5chmidti) [inline clang-tools-extra/test/clang-tidy/checkers/cppcoreguidelines/prefer-member-initializer.cpp:641]: These `CHECK-FIXES` could be written without regex, which would make them more readable.

**Review by 5chmidti — CHANGES_REQUESTED** [review]
Just two nits.

File: clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp [hunk 3]
```diff
@@ -216,6 +217,7 @@ void PreferMemberInitializerCheck::check(
         else {
           ReplaceRange = Init->getInit()->getSourceRange();
         }
+        AddBracket = isa<InitListExpr>(Init->getInit());
```
Comment (5chmidti) [inline clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp:220]: This could be inside the else-branch right above, right?

**Review by 5chmidti — COMMENTED** [review]
~FYI: I have found a false-negative that is unrelated to this pr and issue, but my fix builds upon your pr. I filed an issue for it here #81613 and will post a pr once yours is merged, I don't want to make a stack just for that fix. And adding it to this pr would fix two things in one pr.~

**Review by 5chmidti — APPROVED** [review]

**Review by PiotrZSL — APPROVED** [review]
LGTM

**Pull request merged**